tools/UnitTestApp/ChangeFunctionVisibilityScript.js

```diff
@@ -0,0 +1,74 @@
+const fs = require('fs');
+const path = require('path');
+const { exec } = require('child_process');
+const core = require('@actions/core');
+
+function getPWD(callback) {
+  try {
+    const osType = core.getInput('osType');
+    console.log(`The value of osType is: ${osType}`);
+
+    if (osType === 'windows-latest') {
+      exec('cd', (error, stdout, stderr) => {
+        if (error) {
+          console.error(`exec error: ${error}`);
+          return;
+        }
+        // Remove the trailing newline character
+        const pwd = stdout.trim();
+        // Call the callback function with the result
+        callback(pwd);
+      });
+      return;
+    } else if (osType === 'ubuntu-latest' || osType === 'macos-latest') {
+      exec('echo $PWD', (error, stdout, stderr) => {
```

Comment:
no need to duplicate all the code if only this command is different across OSes.

Reply:
We need the PWD to be passed to the readDirectory function as readDirectory(`${pwd}/repo1/packages/react-charting/src/components/`);

Reply:
in the getPwd function, I see that only the string representing command is different in the if else block.
You can extract it out to say. string pwdCommand = 'cd' if os == windows or '$PWD' otherwise.

Reply:
updated